### ISSUE TITLE
Postgresql support?

### ISSUE THREAD

**Issue by davidworkman9** [joomla-docker/docker-joomla#42]
Is there any support for postgres?

**Comment by mbabker** [comment]
At the moment no but I'd take a pull request adding build variants with PostgreSQL support.

**Comment by J0WI** [comment]
@mbabker this has been fixed in #48.

**Comment by mbabker** [comment]
Well, yes and no.  We need to make sure our entrypoint script doesn't cause issues if you try to boot connected to a PostgreSQL container as that file does a lot of validations for MySQL stuff (the PHP dependencies are loaded now, but the runtime steps need checking).

**Comment by stepankuzmin** [comment]
Hey everyone! If anyone is interested, I've created gist with Docker Compose + Joomla + PostgreSQL + Nginx setup.

https://gist.github.com/stepankuzmin/6b7d206f921221f74086c162f9b6ac24

**Comment by AtjonTV** [comment]
So, no plans to support PostgreSQL anytime soon?

**Comment by Llewellynvdm** [comment]
So first of all thank you @stepankuzmin you configurations does indeed work, even with the latest images, and even with traefik

Here is the details I used to test this:
```
version: '3.7'

services:
  postgress_nginx:
    image: nginx:alpine
    container_name: postgress_nginx
    restart: unless-stopped
    ports:
      - 8080:80
    volumes:
      - "${PROJECT_PATH}/postgress/joomla:/var/www/html"
      - "${PROJECT_PATH}/postgress/nginx.conf:/etc/nginx/nginx.conf"
    depends_on:
      - postgress_joomla
    networks:
      - traefik
    labels:
      # joomla
      - "traefik.enable=true"
      - "traefik.http.routers.postgress_nginx.rule=Host(`postgress.builder.vdm`)"
      - "traefik.http.services.postgress_nginx.loadbalancer.server.port=80"

  postgress_joomla:
    image: joomla:3.10.11-php7.4-fpm-alpine
    container_name: postgress_joomla
    restart: unless-stopped
    environment:
      - JOOMLA_DB_HOST=postgress_db:5432
      - JOOMLA_DB_USER=postgres
      - JOOMLA_DB_PASSWORD=mysecretpassword
      - JOOMLA_DB_NAME=db
    volumes:
      - "${PROJECT_PATH}/postgress/joomla:/var/www/html"
      - "${PROJECT_PATH}/postgress/makedb.php:/makedb.php"
    depends_on:
      - postgress_db
    networks:
      - traefik

  postgress_db:
    image: postgres:latest
    container_name: postgress_db
    restart: unless-stopped
    environment:
      - POSTGRES_DB=db
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=mysecretpassword
    volumes:
      - "${PROJECT_PATH}/postgress/db:/var/lib/postgresql/data"
    networks:
      - traefik

networks:
  traefik:
    external:
      name: traefik_webgateway
```
So we can make some changes to accommodate the _**postgress**_ implementation.

**Comment by Llewellynvdm** [comment]
I did not get around to working on this last year. Except for the official release stuff I have taken a few weeks "break" from all community support. I will make this a high priority soon as I am back, which will be beginning March the soonest.

**Comment by Llewellynvdm** [comment]
Changes has been merged and will become available when we release Joomla 4.3.1

**Comment by Raegedoc** [comment]
Thank you very much for adding official PostgreSQL support to Joomla Docker images.

I had a problem connecting to my external PostgreSQL database and had to fix makedb.php to add " dbname={$db}" to the connection string :

```
elseif ($dbType === 'pgsql')
{
	$port = $port ? (int)$port : 5432;
	$maxTries = 10;

	do {
		$connection = "host={$host} port={$port} **dbname={$db}** user={$user} password={$password}";
		$dbconn = @pg_connect($connection);

```
Am I the only one to have had this problem ?
